### ISSUE TITLE
[BUG] - `cardano-cli ping` fails with `UnknownVersionInRsp` error

### ISSUE THREAD

**Issue by mkoura** [input-output-hk/cardano-node#5324]
*Internal*

**Area**
*Other*   Any other topic (Delegation, Ranking, ...).

**Summary**
Running `cardano-cli ping` fails with `UnknownVersionInRsp` error:

```
$ cardano-cli ping --count 5 --host localhost --port 32005 --magic 42
127.0.0.1:32005 network rtt: 0.000
127.0.0.1:32005 handshake rtt: 0.00027061s
127.0.0.1:32005 Protocol error UnknownVersionInRsp 12
(AddrInfo {addrFlags = [], addrFamily = AF_INET6, addrSocketType = Stream, addrProtocol = 6, addrAddress = [::1]:32005, addrCanonName = Nothing},Network.Socket.connect: <socket: 11>: does not exist (Connection refused))
```

and

```
$ cardano-cli ping --count 5 --unixsock /home/martink/Source/repos/cardano-node/state-cluster0/pool1.socket --magic 42
/home/martink/Source/repos/cardano-node/state-cluster0/pool1.socket network rtt: 0.000
/home/martink/Source/repos/cardano-node/state-cluster0/pool1.socket handshake rtt: 0.000223892s
/home/martink/Source/repos/cardano-node/state-cluster0/pool1.socket Protocol error UnknownVersionInRsp 32784
```

Likely caused by https://github.com/input-output-hk/cardano-node/pull/5313

**System info:**
- OS Name: Fedora 37
- Node version: a7803c24de6a97d72cd025f71631ea65d078c19e
- CLI version: a7803c24de6a97d72cd025f71631ea65d078c19e

**Comment by carbolymer** [comment]
I've tested on a7803c24d :
1. I cannot reproduce an error when using port number:
    ```
    $ cabal run cardano-cli -- ping --magic 2 --count 5 -p 3001 -h localhost
    127.0.0.1:3001 network rtt: 0.000
    127.0.0.1:3001 handshake rtt: 0.000396707s
    127.0.0.1:3001 Negotiated version NodeToNodeVersionV11 2 InitiatorOnly
    timestamp,                      host,                         cookie,  sample,  median,     p90,    mean,     min,     max,     std
    2023-06-15T13:43:11.664821375Z, 127.0.0.1:3001,                    0,   0.001,   0.001,   0.001,   0.001,   0.001,   0.001,     NaN
    2023-06-15T13:43:12.667349748Z, 127.0.0.1:3001,                    1,   0.001,   0.001,   0.001,   0.001,   0.001,   0.001,   0.000
    2023-06-15T13:43:13.66885956Z,  127.0.0.1:3001,                    2,   0.001,   0.001,   0.001,   0.001,   0.001,   0.001,   0.000
    2023-06-15T13:43:14.671362853Z, 127.0.0.1:3001,                    3,   0.001,   0.001,   0.001,   0.001,   0.001,   0.001,   0.000
    2023-06-15T13:43:15.67355866Z,  127.0.0.1:3001,                    4,   0.001,   0.001,   0.001,   0.001,   0.001,   0.001,   0.000
    ````
2. When using socket, I'm getting a different error:
    ```
    ± cabal run cardano-cli -- ping --magic 2 --count 5 --unixsock $CARDANO_NODE_SOCKET_PATH                           
    /home/mgalazyn/workspace/iohk/network-preview/node.socket network rtt: 0.000
    /home/mgalazyn/workspace/iohk/network-preview/node.socket handshake rtt: 0.000449807s
    /home/mgalazyn/workspace/iohk/network-preview/node.socket Protocol error DecodeError 32783 "unknown encoding: TInt 2"
    ```

It is supposed to be fixed in https://github.com/input-output-hk/ouroboros-network/pull/4601 and https://github.com/input-output-hk/cardano-node/pull/5326

**Comment by mkoura** [comment]
Looks to be fixed in both master and 8.1.1.
